packages/manager/modules/billing-components/src/components/commitment/commitment.controller.js

```diff
@@ -104,9 +105,31 @@ export default class {
         (periodic.totalPrice.value / upfront.totalPrice.value - 1) * 100,
       );
       this.savings = periodic.getPriceDiff(upfront);
+      let totalSavings = this.savings.value;
+      totalSavings += this.model.duration.savings
```

Comment:
`totalSavings += this.model.duration.savings?.value || 0`

Reply:
It just a another way of doing it. I don't see any performance improvements.